### ISSUE TITLE
google_access_context_manager_service_perimeter_{ingress|engress}_policy resources get deleted on the second actuation

### ISSUE THREAD

**Issue by mahvar** [hashicorp/terraform-provider-google#17411]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request.
* Please do not leave _+1_ or _me too_ comments, they generate extra noise for issue followers and do not help prioritize the request.
* If you are interested in working on this issue or have submitted a pull request, please leave a comment.
* If an issue is assigned to a user, that user is claiming responsibility for the issue.
* Customers working with a Google Technical Account Manager or Customer Engineer can ask them to [reach out internally](https://github.com/hashicorp/terraform-provider-google/wiki/Customer-Contact#raising-gcp-internal-issues-with-the-provider-development-team)
  to expedite investigation and resolution of this issue.

### Terraform Version

Terraform v1.7.2
on linux_amd64
+ provider registry.terraform.io/hashicorp/google v5.16.0
+ provider registry.terraform.io/hashicorp/google-beta v5.16.0
+ provider registry.terraform.io/hashicorp/null v3.2.2

### Affected Resource(s)

google_access_context_manager_service_perimeter_ingress_policy 
google_access_context_manager_service_perimeter_engress_policy 
google_access_context_manager_service_perimeter

### Terraform Configuration

```tf

resource "google_access_context_manager_service_perimeter" "perimeter" {
  parent                    = "accessPolicies/${local.policy_name}"
  name                      = "accessPolicies/${local.policy_name}/servicePerimeters/${local.perimeter_name}"
  title                         = local.perimeter_name
  perimeter_type     = "PERIMETER_TYPE_REGULAR"

  status {
    resources           = ["projects/999"]
    restricted_services = local.restricted_services
    vpc_accessible_services {
      enable_restriction = true
      allowed_services   = local.allowed_services
    }
  }
}

resource "google_access_context_manager_service_perimeter_egress_policy" "egress_policy" {
  perimeter = google_access_context_manager_service_perimeter.perimeter.id
  egress_from {
    identities = local.lab_bots
  }
  egress_to {
    resources = ["projects/123"]
    operations {
      service_name = "storage.googleapis.com"
      method_selectors {
        method = "google.storage.objects.get"
      }
    }
  }
}
```

### Debug Output

```
google_access_context_manager_service_perimeter.perimeter will be updated in-place

~ resource "google_access_context_manager_service_perimeter" "perimeter" {
id = "blah blah"
name = "blah blah"
# (4 unchanged attributes hidden)

  ~ status {
        # (3 unchanged attributes hidden)

      - egress_policies {
          - egress_from {
              - identities = [
                  - "serviceAccount:blah blah",
                ] -> null
            }
          - egress_to {
              - external_resources = [] -> null
              - resources          = [
                  - "projects/blah blah",
                ] -> null

              - operations {
                  - service_name = "storage.googleapis.com" -> null

                  - method_selectors {
                      - method = "google.storage.objects.get" -> null
                    }
                }
            }
        }

        # (1 unchanged block hidden)
    }
}
```

### Expected Behavior

`terraform apply` shall detect no changes on subsequent runs.

### Actual Behavior

On the first terraform apply, everything works as expected (i.e egress policy gets associated with the perimeter). On the second terraform apply, terraform tries to delete the egress policy.

### Steps to reproduce

1. `terraform init`
2. `terraform apply`
3. `terraform apply`

 **Note**: The second time you try to apply you will notice the issue

### Important Factoids

BTW, When I run terraform apply -refresh-only after the first actuation, it correctly detects the state discrepancy and correctly attaches egress policy to the perimeter resource. However, even after that actual apply still wants to remove the egress policy.

This issue was originally raised against terraform https://github.com/hashicorp/terraform/issues/34729

b/327065086

**Comment by zli82016** [comment]
A diff suppress function is needed for the field `egress_policies` in the resource `google_access_context_manager_service_perimeter` to suppress the diff.

**Comment by mahvar** [comment]
@zli82016 is your comment actionable for a user like me? If yes, I need a little more help.

**Comment by zli82016** [comment]
> @zli82016 is your comment actionable by a user like me? If yes, I need a little more help.

@mahvar , sorry for the confusion. It is a bug in the provider code and my comment is how to fix the bug by changing the code and not actionable by a customer.

**Comment by bradegler** [comment]
@zli82016 any update on when this issue might be fixed? This is a significant blocking issue for my team.

**Comment by zli82016** [comment]
@bradegler , this ticket has been forwarded to the service team and can be tracked in b/327065086

**Comment by sammol-bullish** [comment]
Is this issue fixed yet? Same as @bradegler this is blocking my company to use this resource and make it hard to manage the VPC service control policy among all projects.

**Comment by zli82016** [comment]
loop in the oncall @melinath

**Comment by melinath** [comment]
I'm not able to reproduce this issue with the above information using 6.11.0. I'm going to go ahead and close this for now for clarity; however, if folks are still experiencing the issue it would be great to get a reproducible config or more information about what triggers the diff.

Thanks!

**Comment by melinath** [comment]
(please open a [new issue](https://github.com/hashicorp/terraform-provider-google/issues/new?assignees=&labels=bug&projects=&template=00_bug.yml) if you are experiencing this bug.)
